### PR TITLE
Refacto transaction types

### PR DESCRIPTION
**Pull request by LucasLvy**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Bugfix
- Feature
- Code style update (formatting, renaming)
- Refactoring (no functional changes, no API changes)
- Build-related changes
- Documentation content changes
- Testing
- Other (please describe):

# What is the current behavior?

Resolves: #NA

# What is the new behavior?

-
-
-

# Does this introduce a breaking change?

# Other information

### CODE REVIEW

File: crates/primitives/starknet/src/transaction/types.rs [hunk 1]
```diff
@@ -93,6 +95,135 @@ impl From<TxType> for TransactionType {
...
+    /// Transaction version.
+    pub version: u8,
+    /// Transaction sender address.
+    pub sender_address: ContractAddressWrapper,
+    /// Transaction calldata.
+    pub calldata: BoundedVec<U256, MaxCalldataSize>,
+    /// Account contract nonce.
+    pub nonce: U256,
+    /// Transaction salt.
+    pub salt: U256,
```
Comment (EvolveArt) [inline crates/primitives/starknet/src/transaction/types.rs:209]: salt ?

File: crates/primitives/starknet/src/transaction/types.rs [hunk 2]
```diff
@@ -93,6 +95,135 @@ impl From<TxType> for TransactionType {
...
+    /// Transaction signature.
+    pub signature: BoundedVec<H256, MaxArraySize>,
```
Comment (EvolveArt) [inline crates/primitives/starknet/src/transaction/types.rs:211]: we need a max_fee param

File: crates/primitives/starknet/src/transaction/types.rs [hunk 3]
```diff
@@ -121,9 +252,90 @@ pub struct Transaction {
...
+            calldata: value.call_entrypoint.calldata,
+            salt: value.contract_address_salt.unwrap_or_default(),
+            account_class_hash: value.call_entrypoint.class_hash.ok_or(TransactionConversionError::MissingClassHash)?,
+        })
+    }
+}
+
+impl From<InvokeTransaction> for Transaction {
+    fn from(value: InvokeTransaction) -> Self {
+        // TODO: compute transaction hash
```
Comment (EvolveArt) [inline crates/primitives/starknet/src/transaction/types.rs:275]: issue

File: crates/primitives/starknet/src/transaction/types.rs [hunk 4]
```diff
@@ -93,6 +95,135 @@ impl From<TxType> for TransactionType {
...
+    /// Class hash to declare.
+    pub class_hash: [u8; 32],
+    /// Contract to declare.
+    pub contract_class: ContractClassWrapper,
+    /// Transaction calldata.
+    pub calldata: BoundedVec<U256, MaxCalldataSize>,
+    /// Account contract nonce.
+    pub nonce: U256,
+    /// Transaction salt.
+    pub salt: U256,
```
Comment (EvolveArt) [inline crates/primitives/starknet/src/transaction/types.rs:126]: salt ?

File: crates/primitives/starknet/src/transaction/types.rs [hunk 5]
```diff
@@ -93,6 +95,135 @@ impl From<TxType> for TransactionType {
...
+    /// Transaction signature.
+    pub signature: BoundedVec<H256, MaxArraySize>,
```
Comment (EvolveArt) [inline crates/primitives/starknet/src/transaction/types.rs:128]: todo: add max_fee

File: crates/primitives/starknet/src/transaction/types.rs [hunk 6]
```diff
@@ -93,6 +95,135 @@ impl From<TxType> for TransactionType {
...
+    scale_codec::MaxEncodedLen,
+)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+pub struct DeclareTransaction {
+    /// Transaction version.
+    pub version: u8,
+    /// Transaction sender address.
+    pub sender_address: ContractAddressWrapper,
+    /// Class hash to declare.
+    pub class_hash: [u8; 32],
```
Comment (EvolveArt) [inline crates/primitives/starknet/src/transaction/types.rs:118]: rename to compiled_class_hash

File: crates/primitives/starknet/src/transaction/types.rs [hunk 7]
```diff
@@ -93,6 +95,135 @@ impl From<TxType> for TransactionType {
...
+    /// Contract to declare.
+    pub contract_class: ContractClassWrapper,
+    /// Transaction calldata.
+    pub calldata: BoundedVec<U256, MaxCalldataSize>,
```
Comment (EvolveArt) [inline crates/primitives/starknet/src/transaction/types.rs:122]: no calldata

File: crates/primitives/starknet/src/transaction/types.rs [hunk 8]
```diff
@@ -93,6 +95,135 @@ impl From<TxType> for TransactionType {
...
+    Default,
+    PartialEq,
+    Eq,
+    scale_codec::Encode,
+    scale_codec::Decode,
+    scale_info::TypeInfo,
+    scale_codec::MaxEncodedLen,
+)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+pub struct DeclareTransaction {
```
Comment (EvolveArt) [inline crates/primitives/starknet/src/transaction/types.rs:112]: do we support `DeclareV2` directly ? Or is that V0V1 ?

File: crates/primitives/starknet/src/transaction/types.rs (hunk 5 above)
Comment (LucasLvy) [reply crates/primitives/starknet/src/transaction/types.rs:128]: oh you're right not a todo though

File: crates/primitives/starknet/src/transaction/types.rs (hunk 3 above)
Comment (LucasLvy) [reply crates/primitives/starknet/src/transaction/types.rs:275]: Doing it now

File: crates/primitives/starknet/src/transaction/types.rs (hunk 8 above)
Comment (LucasLvy) [reply crates/primitives/starknet/src/transaction/types.rs:112]: It's V1, v0 is deprecated and we don't support cairo 1 yet

**Review by EvolveArt — APPROVED** [review]
lgtm

**Pull request merged**